pkg/db/sqlc/prune.sql

```diff
@@ -0,0 +1,21 @@
+-- name: CountExpiredEnvelopes :one
+SELECT COUNT(*)::bigint AS expired_count
+FROM public.gateway_envelopes
+WHERE expiry IS NOT NULL
+  AND expiry < EXTRACT(EPOCH FROM now())::bigint;
+
+-- name: DeleteExpiredEnvelopesBatch :many
+WITH to_delete AS (
```

Comment:
If we broke this into two distinct queries we could run the read path on a replica and the deletion on the primary, to better isolate the slow query from prod traffic.

Does get a little messy with replication lag (deleted messages may still appear in subsequent queries).

Not a suggestion. Just a thought.

Reply:
I have not given too much thought to running an HA (or concurrent) version of this. The job is expected to be run as a cron job at some interval. The query is safe to be run concurrently but not optimized for that case.

`CountExpiredEnvelopes` might be an overkill. We will see. I want some diagnostics in the logs, but it is a fairly expensive full table scan. If it becomes too expensive, we can get rid of it.

Reply:
I have not given too much thought to running an HA (or concurrent) version of this. The job is expected to be run as a cron job at some interval. The query is safe to be run concurrently but not optimized for that case.

`CountExpiredEnvelopes` might be an overkill. We will see. I want some diagnostics in the logs, but it is a fairly expensive full table scan. If it becomes too expensive, we can get rid of it.